tests/integration/security/file_mounted_certs/p2p_mtls_test.go

```diff
@@ -0,0 +1,410 @@
+					Set(echo.SidecarVolumeMount, `{"server-certs":{"mountPath":"/server-certs"},"client-certs":{"mountPath":"/client-certs"},"workload-certs":{"mountPath":"/etc/certs"}}`).
+					Set(echo.SidecarConfig, `{"controlPlaneAuthPolicy":"MUTUAL_TLS","proxyMetadata":` + strings.Replace(ProxyMetadataJson, "\n", "", -1) + `}`),
+			}},
+		}).
+		With(&internalServer, echo.Config{
+			Service:   "server",
+			Namespace: appsNamespace,
+			Ports: []echo.Port{
+				{
+					Name:         "http",
+					Protocol:     protocol.HTTP,
+					ServicePort:  8443,
+					InstancePort: 8443,
+					TLS:          false,
+				},
+			},
+			Subsets: []echo.SubsetConfig{{
+				Version:     "v1",
+				// Set up custom annotations to mount the certs.
+				Annotations: echo.NewAnnotations().
+					// workload-certs are needed in order to load the "default" SDS resource, which will be used for the xds-grpc mTLS (tls_certificate_sds_secret_configs.name == "default")
+					// the default bootstrap template does not support reusing values from the `ISTIO_META_TLS_CLIENT_*` environment variables
+					// see security/pkg/nodeagent/cache/secretcache.go:generateFileSecret() for details
+					Set(echo.SidecarVolume, `{"server-certs":{"secret":{"secretName":"test-server-cred"}},"client-certs":{"secret":{"secretName":"test-client-cred"}},"workload-certs":{"secret":{"secretName":"test-istiod-client-cred","items":[{"key":"ca.pem","path":"root-cert.pem"},{"key":"cert.pem","path":"cert-chain.pem"},{"key":"key.pem","path":"key.pem"}]}}}`).
+					Set(echo.SidecarVolumeMount, `{"server-certs":{"mountPath":"/server-certs"},"client-certs":{"mountPath":"/client-certs"},"workload-certs":{"mountPath":"/etc/certs"}}`).
+					Set(echo.SidecarConfig, `{"controlPlaneAuthPolicy":"MUTUAL_TLS","proxyMetadata":` + strings.Replace(ProxyMetadataJson, "\n", "", -1) + `}`),
+			}},
+		}).
+		BuildOrFail(t)
+
+	return internalClient, internalServer, appsNamespace
+}
+
+
+func clusterName(target echo.Instance, port echo.Port) string {
+	cfg := target.Config()
+	return fmt.Sprintf("outbound|%d||%s.%s.svc.%s", port.ServicePort, cfg.Service, cfg.Namespace.Name(), cfg.Domain)
+}
+
+
+func getMetricForCluster(metrics []*dto.Metric, clusterName string) *dto.Metric {
+	for _, metric := range metrics {
+		for _, label := range metric.Label {
+			if (*label.Name == "cluster_name" && *label.Value == clusterName) {
+				return metric
+			}
+		}
+	}
+	return nil
+}
+
+func getMetricForListener(metrics []*dto.Metric, listenerName string) *dto.Metric {
+	for _, metric := range metrics {
+		for _, label := range metric.Label {
+			if (*label.Name == "listener_address" && *label.Value == listenerName) {
+				return metric
+			}
+		}
+	}
+	return nil
+}
+
+
+func verifyUsingFileBasedCerts(t *testing.T, client echo.Instance, server echo.Instance) {
```

Comment:
Won't the test fail with cert mismatch if they aren't? IMO integration tests should be testing behavior - if you want to test the generated XDS output it should be in a unit test

Reply:
In theory it should fail, yes.
But I keep thinking about a scenario when, for some reason, the ISTIO_META_CERT_X params were not recognized and everything is still functioning with the default setup and not using File Mounted Certs.
That's the reason I wanted to have this verification here.